pkg/controller/controller.go

```diff
@@ -522,3 +523,38 @@ func isInstanceReady(instance *v1alpha1.Instance) bool {
 
 	return false
 }
+
+func IsOwnedBy(obj metav1.Object, owner metav1.Object) bool {
+	ref := GetControllerOf(obj)
+	if ref == nil {
+		return false
+	}
+	return ref.UID == owner.GetUID()
+}
+
+// GetControllerOf returns the controllerRef if controllee has a controller,
+// otherwise returns nil.
+func GetControllerOf(controllee metav1.Object) *metav1.OwnerReference {
+	ownerRefs := controllee.GetOwnerReferences()
+	for i := range ownerRefs {
+		owner := &ownerRefs[i]
+		if owner.Controller != nil && *owner.Controller == true {
+			return owner
+		}
+	}
+	return nil
+}
+
+// NewControllerRef creates an OwnerReference pointing to the given owner.
+func NewControllerRef(owner metav1.Object, gvk *schema.GroupVersionKind) *metav1.OwnerReference {
```

Comment:
`GroupVersionKind` - ok
non-pointer - see [deployment](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/sync.go#L649) and [daemon](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemoncontroller.go#L1223)